### PR TITLE
Updating proxy to return 301 to add a "/" at the end for #4958

### LINKED ISSUE #4958: Apiserver proxy requires a trailing slash
This works:
```
 https://130.211.129.169/api/v1beta1/proxy/services/monitoring-grafana/
```
but this does not:
```
 https://130.211.129.169/api/v1beta1/proxy/services/monitoring-grafana
```

We should somehow emphasize the need for the trailing slash for now, and then work out what is going wrong (client side scripting?) to fix this (our fault in the proxy or Grafana's?)

@vishh

### PR DESCRIPTION
**Pull request by nikhiljindal**
Fixes https://github.com/GoogleCloudPlatform/kubernetes/issues/4958

cc @vishh @satnam6502

### CODE REVIEW

**Comment by nikhiljindal** [comment]
@lavalamp Note that I had to change TestProxyUpgrade() test. I dont understand web socket case well. I hope I am not breaking it by this change.

**Comment by vishh** [comment]
Thanks for the PR @nikhiljindal. It helps improve user experience!

**Comment by lavalamp** [comment]
I'd suggest moving the redirect below the call to tryUpgrade() so that you don't have to change the upgrade test-- not sure if clients that use that will be able to follow a 301.

Also can you make your @google.com email address public on your github profile, I think that might satisfy CLA bot?

**Comment by nikhiljindal** [comment]
Done. Moved the code to after tryUpgrade().

Also, my @google.com email address is my primary email and its public on my profile already. Let me know if there is something else I can go to satisfy the CLA bot.

File: pkg/apiserver/proxy_test.go [hunk 1]
```diff
@@ -367,3 +367,58 @@ func TestProxyUpgrade(t *testing.T) {
...
+		req, err := http.NewRequest(
+			"GET",
+			server.URL+proxyTestPattern,
+			strings.NewReader(""),
+		)
+		if err != nil {
+			t.Errorf("unexpected error %v", err)
+			continue
+		}
+		resp, err := http.DefaultClient.Do(req)
```
Comment (lavalamp) [inline pkg/apiserver/proxy_test.go]: Can you add a comment here to the effect that the default client follows the redirect, which is why this test works? Took me a minute to figure it out. :)

**Comment by lavalamp** [comment]
LGTM, one nit

File: pkg/apiserver/proxy_test.go (hunk 1 above)
Comment (nikhiljindal) [inline pkg/apiserver/proxy_test.go]: Done.

**Comment by lavalamp** [comment]
LGTM

**Comment by lavalamp** [comment]
I don't understand why, since it passed before I merged, but this seems to have broken head. Reverting.

**Pull request merged**